DataDog/datadog-operator
# [CECO-1067] DDA config for Admission Controller sidecar injection using selector and profile

### What does this PR do?

This PR is following up PR #1207 adding selectors and profiles.


### Motivation

CECO-1067

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

Test based on the release v1.6.0
- Run DDA with `selectors.objectSelector` option in admissionController.agentSidecarInjection.
- Test `selectors.namespaceSelector` with same structure.

```
  features:
        selectors:
        - objectSelector:
            matchLabels:
              "testLabel": "testValue"
```
```
  features:
        selectors:
        - namespaceSelector:
            matchLabels:
              "testLabel": "testValue"
```

- Check that the `DD_ADMISSION_CONTROLLER_AGENT_SIDECAR_SELECTORS` is set to correct map structure.
- Run command : kubectl describe deploy datadog-cluster-agent | grep -i DD_ADMISSION_CONTROLLER_AGENT_SIDECAR


```
Result #1
      DD_ADMISSION_CONTROLLER_AGENT_SIDECAR_SELECTORS:              [{"objectSelector":{"matchLabels":{"testLabel":"testValue"}}}]
Result #2
      DD_ADMISSION_CONTROLLER_AGENT_SIDECAR_SELECTORS:              [{"namespaceSelector":{"matchLabels":{"testLabel":"testValue"}}}]

```

- Next, test DDA with `profiles` option in admissionController.agentSidecarInjection.

```
  features:
    admissionController:
      agentSidecarInjection:
        enabled: true
        provider: "fargate"
        profiles:
        - env:
          - name: DD_PROCESS_AGENT_PROCESS_COLLECTION_ENABLED
            value: "true"
          resources:
            requests:
              cpu: "400m"
              memory: "256Mi"
            limits:
              cpu: "800m"
              memory: "512Mi"
```

- Check that the `DD_ADMISSION_CONTROLLER_AGENT_SIDECAR_PROFILES` is set to correct map structure.
- Run command : kubectl describe deploy datadog-cluster-agent | grep -i DD_ADMISSION_CONTROLLER_AGENT_SIDECAR

```
Result : 
     DD_ADMISSION_CONTROLLER_AGENT_SIDECAR_PROFILES:               [{"env":[{"name":"DD_PROCESS_AGENT_PROCESS_COLLECTION_ENABLED","value":"true"}],"resources":{"limits":{"cpu":"800m","memory":"512Mi"},"requests":{"cpu":"400m","memory":"256Mi"}}}]

```

### Checklist
